awswrangler/neptune/neptune.py

```diff
@@ -0,0 +1,134 @@
+
+
+    """
+    raise NotImplementedError
+
+def read_opencypher(
+    client: client,
+    traversal: str
+) -> pd.DataFrame:
+    """Return results of a openCypher traversal as pandas dataframe.
+
+    Parameters
+    ----------
+    client : neptune.Client
+        instance of the neptune client to use
+    traversal : str
+        The gremlin traversal to execute
+
+    Returns
+    -------
+    Union[pandas.DataFrame, Iterator[pandas.DataFrame]]
+        Results as Pandas DataFrame
+
+    Examples
+    --------
+    Run an openCypher query
+
+    >>> import awswrangler as wr
+    >>> client = wr.neptune.Client(host='NEPTUNE-ENDPOINT')
+    >>> df = wr.neptune.gremlin.read(client, "MATCH (n) RETURN n LIMIT 5")
+
+
+    """
+    raise NotImplementedError
+
+def read_sparql(
+    client: client,
+    traversal: str
+) -> pd.DataFrame:
+    """Return results of a SPARQL query as pandas dataframe.
+
+    Parameters
+    ----------
+    client : neptune.Client
+        instance of the neptune client to use
+    traversal : str
+        The gremlin traversal to execute
+
+    Returns
+    -------
+    Union[pandas.DataFrame, Iterator[pandas.DataFrame]]
+        Results as Pandas DataFrame
+
+    Examples
+    --------
+    Run a SPARQL query
+
+    >>> import awswrangler as wr
+    >>> client = wr.neptune.Client(host='NEPTUNE-ENDPOINT')
+    >>> df = wr.neptune.sparql.read(client, "PREFIX foaf:  <http://xmlns.com/foaf/0.1/>
+    SELECT ?name
+    WHERE {
+            ?person foaf:name ?name .
+    }")
+    """
+    raise NotImplementedError
+
+def to_graph(
+    client: client,
+    df: pd.DataFrame
+) -> None:
+    """Write records stored in a DataFrame into Amazon Neptune.    
+    
+    If writing to a property graph then DataFrames for vertices and edges must be written as separetly. 
+    DataFrames for vertices must have a ~label column with the label and a ~id column for the vertex id.  
+    If the ~id column does not exist, the specified id does not exists, or is empty then a new vertex will be added.  
+    If no ~label column exists an exception will be thrown.  
+    DataFrames for edges must have a ~id, ~label, ~to, and ~from column.  If the ~id column does not exist, 
+    the specified id does not exists, or is empty then a new edge will be added. If no ~label, ~to, or ~from column exists an exception will be thrown.  
+
+    If writing to RDF then the DataFrame must consist of triples with column names of s, p, and o.
+
+    Parameters
+    ----------
+    client : neptune.Client
+        instance of the neptune client to use
+    df : pandas.DataFrame
+        Pandas DataFrame https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html
+
+    Returns
+    -------
+    None
+        None.
+
+    Examples
+    --------
+    Writing to Amazon Neptune 
+
+    >>> import awswrangler as wr
+    >>> client = wr.neptune.Client(host='NEPTUNE-ENDPOINT')
+    >>> wr.neptune.gremlin.to_graph(
```

Comment:
Would there be one per graph type (i.e. gremlin, sparql) or should this be `wr.neptune.to_graph`?

Reply:
After thinking about the feedback here, I ended up creating a `to_property_graph()` and `to_rdf_graph()` as the incoming data frame for each will have a different format.  For Gremlin versus openCypher they will have the same format, so I decided to combine them.

Reply:
That would be fine. Once we start implementing these methods it will become clearer if they should merged or not

Reply:
A single method should be fine as long as input is a DataFrame but yes that would become clear when implementation is added